tools/report-converter/codechecker_report_converter/analyzers/analyzer_result.py

```diff
@@ -147,7 +147,7 @@ def _write(
 
         analyzer_info = AnalyzerInfo(name=self.TOOL_NAME)
         for file_path, file_reports in file_to_report.items():
-            source_file = os.path.basename(file_path)
+            source_file = ("-".join(file_path.split("/")[-3:]))
```

Comment:
This feels like too much of a symptom fix and a band-aid, which just begs to be broken sometime later by someone whose directories only differ 3 or 4 hops up the chain.

I'm not sure at a glance how hot this path is, but maybe instead we should put a sort of hash of the full file path into the file name?

(At least that is what clangd does with its cache: `Something.cpp.AAAAAAAAAA.idx` is the file they create, where `AAAA...` is a hash of the configuration that produced the file (so it includes not only path, but a hash of the compile command too, as far as I know.)

Reply:
I agree with @whisperity, this feels like a hacky solution which also not always works.

Now the `report-converter` tool has a `--filename` option which handles two special values: `{source_file}` and `{analyzer}`. My recommendation is to create another special value (e.g.: `{hash}`) which will insert a hash into the file name. 
You can create the file hash similarly to this:
https://github.com/Ericsson/codechecker/blob/2c49428aa4cf86e1e28aef349ef8de27a4382876/analyzer/codechecker_analyzer/analyzers/result_handler_base.py#L132

And the user will be able to use it like this: `report-converter --filename "{source_file}_{analyzer}_{hash}"`

Also it would be good to add a test case for this use case too to test that report-converter will create multiple plist files not just one.